### PR TITLE
Removed V1 support

### PR DESCRIPTION
**Pull request by ewoutp**
This change updates the cli for the removal of V1 support in user-config, client-go and app-def-transf-go.

Dependencies:

- [x] giantswarm/app-def-transf-go#19
- [x] giantswarm/client-go#74

### CODE REVIEW

**Comment by ewoutp** [comment]
@ZeissS RFR

File: int-tests/010-create-invalid.t [hunk 1]
```diff
@@ -2,13 +2,25 @@ Create invalid apps:
 
   $ # Try to create an app with an invalid field at root level
   $ swarm create ${TESTDIR}/apps/invalid/root1.json
-  Cannot parse service config: unknown JSON field: ["bogus"]
+  Creating '' in the 'cli-testbot/dev' environment...
+  Cannot create service due to a validation error in '/.*/cli/int-tests/apps/invalid/root1.json'. (re)
+  
+      Validation error: invalid service definition, expecting valid types of V2GiantSwarm
```
Comment (ZeissS) [inline int-tests/010-create-invalid.t]: thats not an error we should expose to users.

Comment (ewoutp) [inline int-tests/010-create-invalid.t]: I'll create an issue for that

Comment (ewoutp) [inline int-tests/010-create-invalid.t]: See #408

File: Makefile [hunk 2]
```diff
@@ -49,12 +49,12 @@ deps:
...
 	@builder get dep -b 0.6.1 git@github.com:giantswarm/api-schema.git $(GS_PATH)/api-schema
-	@builder get dep -b 0.4.2 git@github.com:giantswarm/app-def-transf-go.git $(GS_PATH)/app-def-transf-go
-	@builder get dep -b 0.29.0 git@github.com:giantswarm/client-go.git $(GS_PATH)/client-go
+	@builder get dep -b 0.5.0 git@github.com:giantswarm/app-def-transf-go.git $(GS_PATH)/app-def-transf-go
+	@builder get dep -b 0.30.0 git@github.com:giantswarm/client-go.git $(GS_PATH)/client-go
 	@builder get dep -b 1.3.2 git@github.com:giantswarm/generic-types-go.git $(GS_PATH)/generic-types-go
 	@builder get dep git@github.com:giantswarm/go-tld.git $(GS_PATH)/go-tld
 	@builder get dep -b 0.1.2 git@github.com:giantswarm/hijack-stream-support.git $(GS_PATH)/hijack-stream-support
-	@builder get dep -b 0.27.0 git@github.com:giantswarm/user-config.git $(GS_PATH)/user-config
+	@builder get dep -b 0.28.0 git@github.com:giantswarm/user-config.git $(GS_PATH)/user-config
```
Comment (ZeissS) [inline Makefile]: Can you update this to `0.28.1`? If I would patch this myself, we would endup with breaking master.

**Comment by ZeissS** [comment]
LGTM

File: Makefile (hunk 2 above)
Comment (ewoutp) [inline Makefile]: Done

**Pull request merged**